Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
+
+	void ScenarioGUI::UpdateHoveredScene(int mouseX, int mouseY) {
+		bool foundAnyHover = false;
+		if (m_ScenarioScenes && !m_DraggedBox && !m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->PointInside(mouseX, mouseY) && !m_ScenarioCollectionBoxes.at(ScenarioCollections::SceneInfoBox)->PointInside(mouseX, mouseY)) {
+			Scene *candidateScene = nullptr;
+			float shortestDist = 10.0F;
+			Vector mousePos(static_cast<float>(mouseX), static_cast<float>(mouseY));
+			for (Scene *scenarioScene : *m_ScenarioScenes) {
+				float distance = (m_PlanetCenter + scenarioScene->GetLocation() + scenarioScene->GetLocationOffset() - mousePos).GetMagnitude();
+				if (distance < shortestDist) {
+					shortestDist = distance;
+					candidateScene = scenarioScene;
+					foundAnyHover = true;
+				}
+			}
+
+			// Set new hovered scene to be the one closest to the cursor, if there is any and if it is different than the currently hovered one.
+			if (candidateScene && candidateScene != m_HoveredScene) {
+				m_HoveredScene = candidateScene;
+				g_GUISound.SelectionChangeSound()->Play();
+
+				m_SitePointLabel->SetText(m_HoveredScene->GetPresetName());
+
+				Vector labelAbsPos = m_PlanetCenter + Vector(m_HoveredScene->GetLocation() + m_HoveredScene->GetLocationOffset()) - Vector(m_SitePointLabel->GetWidth() / 2, 0.0F) - Vector(0.0F, m_SitePointLabel->GetHeight() * 1.5F);
+				float padding = 6.0F;
+				labelAbsPos.m_X = std::clamp(labelAbsPos.m_X, padding, g_FrameMan.GetResX() - m_SitePointLabel->GetWidth() - padding);
+				labelAbsPos.m_Y = std::clamp(labelAbsPos.m_Y, padding, g_FrameMan.GetResY() - m_SitePointLabel->GetHeight() - padding);
+				m_SitePointLabel->SetPositionAbs(labelAbsPos.GetFloorIntX(), labelAbsPos.GetFloorIntY());
+
+				m_SitePointLabel->SetVisible(true);
+			}
+		}
+		if (!foundAnyHover) {
+			m_HoveredScene = nullptr;
+			m_SitePointLabel->SetVisible(false);
 		}
 	}
 
-	// Add offsets to reveal overlapping scenes if any
-    for (list<Scene *>::iterator pItr = filteredScenes.begin(); pItr != filteredScenes.end(); ++pItr)
-    {
-		bool isOverlapped = false;
-
-		do 
-		{
-			isOverlapped = false;
-
-			// Find overlapping scene dot
-			for (list<Scene *>::iterator pItr2 = filteredScenes.begin(); pItr2 != filteredScenes.end(); ++pItr2)
-			{
-				if ((*pItr) != (*pItr2))
-				{
-					Vector pos1 = (*pItr)->GetLocation() + (*pItr)->GetLocationOffset();
-					Vector pos2 = (*pItr2)->GetLocation() + (*pItr2)->GetLocationOffset();
-
-					if ((pos1 - pos2).GetMagnitude() < 8)
-					{
-						isOverlapped = true;
-						break;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	ScenarioGUI::ScenarioUpdateResult ScenarioGUI::UpdateInput() {
+		// TODO: if activity is running, allow esc to resume activity instead of quitting.
```

Comment:
Wanna just do this TODO? Or maybe it's been done already/in some other work? Seems like it should be pretty simple, and it's nice to clean up TODOs (or turn them into issues if they're big) when something is refactored.